cardano-mempool/tests/Test/Cardano/Memory/PoolSpec.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Test.Cardano.Memory.PoolSpec (spec) where
```

Comment:
Some additional things I'd like to see tested:

- Calling finalizers explicitly (`finalizeForeignPtr`)
- Interleaving deallocations with allocations (as discussed, this is more likely to hit code paths where pages get fragmented, and blocks from within an existing page are returned)

Reply:
Totally agree. I'll try to come up with a few tests like that

Reply:
Done. See new `propPoolAllocateAndFinalize` property